### ISSUE TITLE
Add Faugus Launcher as an option in ProtonUp-Qt

### ISSUE THREAD

**Issue by Faugus** [DavidoTek/ProtonUp-Qt#461]
Hello. I'm the developer of [Faugus Launcher](https://github.com/Faugus/faugus-launcher).
It's an app for running Windows games using UMU-Launcher.

Could you please add Faugus Launcher as an option in ProtonUp-Qt?
It grabs Proton builds installed in `~/.local/share/Steam/compatibilitytools.d/`.

Thank you.

**Comment by DavidoTek** [comment]
Hi, thanks for reaching out.

> It's an app for running Windows games using UMU-Launcher.

We had a conversion related to umu and I think Lutris already. It's nice to see that there is now a standalone launcher for it. https://github.com/DavidoTek/ProtonUp-Qt/issues/191#issuecomment-2058037459

> It grabs Proton builds installed in ~/.local/share/Steam/compatibilitytools.d/.

The path `~/.local/share/Steam/compatibilitytools.d/` is detected by default. If the folder (or a symlink) exists, it will be displayed as a "Steam" launcher. That means installing compatibility tools for Faugus is already possible now.

Out of curiosity: Is the Faugus Launcher aware of the other possible locations for the Steam root folder?
https://github.com/DavidoTek/ProtonUp-Qt/blob/d7c71a8e0b6f37a9a1832cd89a8d9c8fdccc6cf0/pupgui2/constants.py#L41-L43

> Could you please add Faugus Launcher as an option in ProtonUp-Qt?

Are there any advanced features like a game list (i.e., mapping of which game uses which compatibility tool) that should be included in ProtonUp-Qt and that aren't handled by the Steam launcher?
If yes, what are the features, where does Faugus store the configuration files and how does the format of the config look like?

What I found is the "runner" configuration: https://github.com/Faugus/faugus-launcher/blob/c1e272d980c45689423a97789f63739f2580b34a/faugus-launcher.py#L2664-L2673

![grafik](https://github.com/user-attachments/assets/5899686a-68e7-407e-bb85-5d67d5782189)

**Comment by Faugus** [comment]
> Is the Faugus Launcher aware of the other possible locations for the Steam root folder?

It's not. It uses only the folder where UMU-Launcher downloads GE-Proton and UMU-Proton latest builds, which is `~/.local/share/Steam/compatibilitytools.d/`.

> Are there any advanced features like a game list

There's not.
The idea is to download specific GE-Proton builds, cause I'm not sure exactly what Proton builds are supported by UMU-Launcher, except GE-Proton and UMU-Proton.
Proton-TKG is not supported according to the devs:
https://github.com/Open-Wine-Components/umu-launcher/issues/99

**Comment by sonic2kk** [comment]
If Faugus Launcher piggybacks off of the Steam install directory, that suggests and expectation that Steam will be installed, right?

If so, why does Faugus need to be added to ProtonUp-Qt, if ProtonUp-Qt can already download to `~/.local/share/Steam/compatibilitytools.d` (or the other Steam Native paths, whatever exists and is not a symlink to an already discovered Steam Native path), would Faugus already detect compatibility tools downloaded by ProtonUp-Qt for Steam?

Sorry if I'm misunderstanding, I would just like to know the functionality that would be provided here, as it sounds like ProtonUp-Qt can already download to this folder.

**Comment by Faugus** [comment]
> that suggests and expectation that Steam will be installed, right?

UMU-Launcher doesn't need Steam to be installed. It just uses the same folder to install the lastest GE-Proton and UMU-Proton.

I'm just asking to put Faugus Launcher as an option, because the user doesn't know that downloading a GE-Proton for Steam is the same thing that downloading for Faugus Launcher.

**Comment by sonic2kk** [comment]
> UMU-Launcher doesn't need Steam to be installed. It just uses the same folder to install the lastest GE-Proton and UMU-Proton.

I know UMU-Launcher doesn't, I was referring to Faugus launcher. It seems like it will use `~/.local/share/Steam/compatibilitytools.d` as its directory to source compatibility tools. ProtonUp-Qt can already download to this folder with Steam.

> I'm just asking to put Faugus Launcher as an option, because the user doesn't know that downloading a GE-Proton for Steam is the same thing that downloading for Faugus Launcher.

The same could be said for Lutris, but I think this is something to document on the launcher side rather than to add another launcher. That's just my opinion though.

**Comment by Faugus** [comment]
> under normal circumstances this folder only exists if Steam creates it

Faugus Launcher sets UMU-Launcher to use GE-Proton Latest by default, but the user can select UMU-Proton Latest as well.
On both circunstances, when the user runs a game, UMU-Launcher creates the `~/.local/share/Steam/compatibilitytools.d` and downloads the selected runner.

**Comment by sonic2kk** [comment]
I suppose then adding Faugus would cover a case where it is installed, and `~/.local/share/Steam/compatibilitytools.d` exists (presumably created by UMU-Launcher in this case), but Steam is not installed.

I do have some questions about why UMU-Launcher does this, and what implications this has if Steam is not installed but is later installed, and also why this path which is not guaranteed to be the Steam `compatibilitytools.d` location, but none of those are centrally relevant to Faugus.

Since `~/.local/share/Steam/compatibilitytools.d` sounds like it's the only install directory that is used - as this is the one created by UMU Launcher - and that Faugus only uses this directory because it is the one created by UMU Launcher, then I guess it _is_ distinct from Steam (Native), which could have a few different install paths.

**Comment by Faugus** [comment]
I don't have Steam installed and it shows the GE and UMU Protons downloaded by UMU-Launcher:
![image](https://github.com/user-attachments/assets/c000dec0-0aea-41ca-9fe6-dffc03faf62c)

**Comment by sonic2kk** [comment]
I'm interested in why ProtonUp-Qt picks it up if the only thing in `~/.local/share/Steam` is the `compatibilitytools.d` folder.

If ProtonUp-Qt can already detect download into this folder, then I am again not sure why Faugus needs to be added explicitly... Sure, users may not know that Faugus / UMU use this, but this *is* a Steam folder, and the tools should document this (whether this be Faugus, UMU upstream, or both, that's a different matter).

**Comment by Faugus** [comment]
Oh my bad.
I had Steam installed before, but not anymore.
There were files and folders left in `~/.local/share/Steam`.
Once I deleted them, leaving only the `compatibilitytools.d`, it's not showing anymore.

**Comment by Faugus** [comment]
I decided to make a bult in tool to manage GE-Proton builds.

Thank you very much for your patience.

**Comment by DavidoTek** [comment]
> There were files and folders left in ~/.local/share/Steam.
Once I deleted them, leaving only the compatibilitytools.d, it's not showing anymore.

ProtonUp-Qt only picks up `~/.local/share/Steam` if the paths `./config/config.vdf` and `./config/libraryfolders.vdf` exist.
If we decided to add Fangus Launcher to ProtonUp-Qt, it would need to skip that check.
As we don't want to display Fangus if only Steam and not Fangus Launcher is installed, there needs to be an additional check whether `~/.config/fangus-launcher/config.ini` exists.
The launcher entry could be called `Fangus / Steam Native` so the user get a hint that they share the same folder.

> I decided to make a built in tool to manage GE-Proton builds.
Thank you very much for your patience.

That's another possibility.
You're welcome, thanks for sharing.
